### PR TITLE
create login template #14

### PR DESCRIPTION
**Pull request by ZinabHashem**

### CODE REVIEW

File: ui/src/app/auth/login/login.component.html [hunk 1]
```diff
@@ -1 +1,41 @@
-<p>login works!</p>
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.html:1]: remove this line

File: ui/src/app/auth/login/login.component.html [hunk 2]
```diff
@@ -1 +1,41 @@
...
+<div fxLayoutAlign="center center" style="margin-top: 3%;">
+  <div fxFlex="95" fxFlex.lt-md="100" fxFlexFill fxLayout="row" style="border: 1px solid rgb(221, 219, 219);  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);">
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.html:2]: do not put inline style in the html

File: ui/src/app/auth/login/login.component.html (hunk 1 above)
Comment (walaayousef98) [reply ui/src/app/auth/login/login.component.html:1]: This line is already removed

**Comment by walaayousef98** [comment]
check changes

File: ui/src/app/auth/login/login.component.html (hunk 2 above)
Comment (ZinabHashem) [reply ui/src/app/auth/login/login.component.html]: ok, modified

File: ui/src/app/auth/login/login.component.ts [hunk 3]
```diff
@@ -7,9 +8,27 @@ import { Component, OnInit } from '@angular/core';
...
+  constructor(private fb:FormBuilder) { }
 
   ngOnInit(): void {
+    this.loginForm = this.fb.group({
+      'email': ['', Validators.compose([Validators.required, Validators.pattern(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/)])],
+      'password': ['', Validators.compose([Validators.required])]
+  });
+  }
+
+  onSubmit(Form:any) {
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.ts:23]: do not use any

File: ui/src/app/auth/login/login.component.ts [hunk 4]
```diff
@@ -7,9 +8,27 @@ import { Component, OnInit } from '@angular/core';
...
+      this.formSubmitted = true;
+      if (this.loginForm.valid) {
+          console.log("success login")
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.ts:26]: do not use console.log

File: ui/src/app/auth/login/login.component.ts (hunk 3 above)
Comment (diaahamad) [reply ui/src/app/auth/login/login.component.ts:23]: and u do not need to pass the from that it is already exist in the class

File: ui/src/app/auth/login/login.component.ts [hunk 5]
```diff
@@ -7,9 +8,25 @@ import { Component, OnInit } from '@angular/core';
...
   ngOnInit(): void {
+    this.loginForm = this.fb.group({
+      'email': ['', Validators.compose([Validators.required, Validators.pattern(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/)])],
+      'password': ['', Validators.compose([Validators.required])]
+  });
+  }
+
+  onSubmit() {
+      if (this.loginForm.valid) {
+        this.formSubmitted = true;
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.ts:25]: by this way the user unable to see the errors, this should be in the first

File: ui/src/app/auth/login/login.component.html [hunk 6]
```diff
@@ -1 +1,43 @@
...
+                <mat-label>Email</mat-label>
+                <input formControlName="email" type="email" matInput  [class.is-invalid]="loginForm.controls['email'].invalid && loginForm.controls['email'].touched" [class.is-valid]="loginForm.controls['email'].valid" required>
+                <mat-error *ngIf="loginForm.controls['email'].invalid">Please enter a valid email</mat-error>
+              </mat-form-field>
+            </div>
+            <div fxLayoutAlign="center">
+              <mat-form-field fxFlex="60" fxFlex.lt-md="90" appearance="outline">
+                <mat-label>Password</mat-label>
+                <input formControlName="password" type="password" matInput [class.is-invalid]="loginForm.controls['password'].invalid && loginForm.controls['password'].touched" [class.is-valid]="loginForm.controls['password'].valid" required>
+                <mat-error *ngIf="loginForm.controls['password'].invalid">Password is required</mat-error>
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.html:29]: in which status those errors will be visible ?

Comment (ZinabHashem) [reply ui/src/app/auth/login/login.component.html:29]: when the user not enter thing.

File: ui/src/app/auth/login/login.component.ts (hunk 5 above)
Comment (walaayousef98) [reply ui/src/app/auth/login/login.component.ts:25]: Why did the user not see the errors?

Comment (ZinabHashem) [reply ui/src/app/auth/login/login.component.ts:25]: Done

**Pull request merged**